sys/include/suit/transport/worker.h

```diff
@@ -39,6 +39,42 @@ extern "C" {
  */
 void suit_worker_trigger(const char *url, size_t len);
 
+/**
+ * @brief   Trigger a SUIT udate via a worker thread
+ *
+ * @pre The caller called @ref suit_worker_prepare to obtain the @p buf, and
+ * populated @p size bytes into it.
+ *
+ * This can be called with a size of 0 when writing a manifest was started, but
+ * could not be completed.
+ *
+ * @param[in] manifest  Pointer to the manifest. This must be the return value
+ *                      of a previous call to @ref suit_worker_prepare, and is
+ *                      invalidated at some point during or after the call.
+ * @param[in] size      Number of bytes that have been prepared in @ref
+ *                      the buffer before the call.
+ */
+void suit_worker_trigger_prepared(const uint8_t *manifest, size_t size);
+
+/**
+ * @brief    Prepare for a worker run with a preloaded manifest
+ *
+ * This obtains a lock on the SUIT worker, and returns a pointer to a memory
+ * area into which the manifest is to be written. The lock must be released by
+ * calling @ref suit_worker_trigger_prepared later.
+ *
+ * @return a pointer on success.
+ * @return NULL if the manifest buffer is currently unavailable, or can not
+ *     accommodate the requested size.
+ *
+ * @note There is no blocking version of this (it behaves like @ref
+ * mutex_trylock, not like @ref mutex_lock). This allows a simpler
+ * implementation on the thread handling side<!-- possibly it's not even
+ * possible without priority inversion tricks -->, and is also what typical use
+ * cases need.
+ */
+uint8_t *suit_worker_prepare(size_t size);
```

Comment:
If the user of this function is about to receive the manifest, they don't know the size yet - so what are they supposed to put here? 

`SUIT_MANIFEST_BUFSIZE` is not even public, so they can only guess.

How about `size_t suit_worker_prepare(void **buf)` - then it's clear we can read up to [return value] bytes into the provided buffer.

Reply:
There *is* the issue that "can't give you that right now" is indistinguishable from "can't give you that ever", which CoAP should represent as 5.03 Service Unavailable (try again in 60 seconds) and 4.13 Request Entity Too Large -- but I'm not sure how that'd best be expressed.

Maybe in a `size_t suit_worker_prepare(size_t size, uint8_t **ptr);` -- where 0 indicates success, a positive number indicates that something of that size would work, and a negative number is -EAGAIN and tells we can't hand out anything right now? It'd be unusual error handling, but I have no better ideas. Do you have any better suggestions?

Reply:
The latest update (https://github.com/RIOT-OS/RIOT/pull/19601/commits/800446b59b0c46dbbcaad836f56b119c40c2e451) now provides better error handling at the cost of two return-by-pointer arguments, and uses a `int suit_worker_prepare(uint8_t **buffer, size_t *size)` signature.

If an application can not at all predict what it will get and needs to take the maximum available, it could now attempt suit_worker_prepare with the maximum size, and try again once it learns a good size. I wouldn't recommend that, though, which is why I'm fully OK with that particular workflow being clumsy.